frontend/packages/console-shared/src/hooks/deep-compare-memoize.ts

```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+
+export const useDeepCompareMemoize = (value) => {
```

Comment:
```suggestion
export const useDeepCompareMemoize<T = any> = (value: T): T => {
```

That way it is clear whatever you're giving is being returned from this method. If they want to type it, they can.

I do see later that this is a move of code, but I'd still like this done.

Reply:
The above suggestion throws error saying parsing error. I think it should be somthing like
```suggestion
export const useDeepCompareMemoize = <T = any>(value: T): T => {
```

But this has issues with type `T` not matching with `ref.current`. Not sure how to correctly type it this way.

Reply:
Ah sorry, I didn't test it to see the issue. I just made the comment.

React Hooks have generics built into them.

```
export const useDeepCompareMemoize = <T = any>(value: T): T => {
  const ref = React.useRef<T>();

  if (!_.isEqual(value, ref.current)) {
    ref.current = value;
  }

  return ref.current;
};
```